tests/test_examples.py

```diff
@@ -1186,6 +1191,333 @@ def test_example_get_current(self):
         )
 
 
+@pytest.mark.regression
+@pytest.mark.solve
+@pytest.mark.slow
+def test_regcoil_axisymmetric():
+    """Test axisymmetric regcoil solution."""
+    # make a simple axisymmetric vacuum equilibrium
+    eq = load("./tests/inputs/vacuum_circular_tokamak.h5")
+    # no phi_SV is needed since it is axisymmetric,
+    # so phi_mn should be zero when running REGCOIL
+    # especially with a nonzero alpha
+    surf_winding = FourierRZToroidalSurface.constant_offset_surface(eq.surface, 2)
+
+    surface_current_field = FourierCurrentPotentialField.from_surface(
+        surf_winding, M_Phi=1, N_Phi=1, sym_Phi="sin"
+    )
+    surface_current_field, data = run_regcoil(
+        surface_current_field,
+        eq,
+        alpha=0,
+        normalize=False,
+        vacuum=True,
+        verbose=2,
+    )
+    chi_B = data["chi^2_B"]
+    phi_mn_opt = surface_current_field.Phi_mn
+    G = surface_current_field.G
+    np.testing.assert_allclose(phi_mn_opt, 0, atol=1e-6)
+    np.testing.assert_allclose(chi_B, 0, atol=1e-10)
+    coords = eq.compute(["R", "phi", "Z", "B"])
+    B = coords["B"]
+    coords = np.vstack([coords["R"], coords["phi"], coords["Z"]]).T
+    B_from_surf = surface_current_field.compute_magnetic_field(
+        coords, source_grid=LinearGrid(M=200, N=200, NFP=surf_winding.NFP)
+    )
+    np.testing.assert_allclose(B, B_from_surf, atol=1e-4)
+
+    grid = LinearGrid(N=10, M=10, NFP=surface_current_field.NFP)
+    correct_phi = G * grid.nodes[:, 2] / 2 / np.pi
+    np.testing.assert_allclose(
+        surface_current_field.compute("Phi", grid=grid)["Phi"], correct_phi, atol=5e-9
+    )
+    surface_current_field.change_Phi_resolution(
+        M=2,
+        N=2,
+    )
+    # test with alpha large, should have no phi_mn
+    surface_current_field, data = run_regcoil(
+        surface_current_field,
+        eq=eq,
+        eval_grid=LinearGrid(M=10, N=10, NFP=eq.NFP, sym=eq.sym),
```

Comment:
Are there cases where eval_grid  is not stellarator symmetric?

Reply:
Yea, if the equilibrium is not stellarator symetric, or if the surface the coils lie on is not stellarator symmetric, or is not symetric wrt the same axis/plane as the equilibrum is